File: app/src/main/java/jp/co/geisha/itunestracker/service/view/activity/VideoViewActivity.java
package jp.co.geisha.itunestracker.service.view.activity;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import jp.co.geisha.itunestracker.R;
import jp.co.geisha.itunestracker.service.util.FragmentUtils;
import jp.co.geisha.itunestracker.service.view.fragment.ArtistVideoViewFragment;

import static jp.co.geisha.itunestracker.service.view.activity.MainActivity.URL;

public class VideoViewActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_video_view);

        ArtistVideoViewFragment fragment = new ArtistVideoViewFragment();

        Intent i = getIntent();
        String artistPreviewUrl = i.getStringExtra(URL);

        addVideoViewFragment(fragment, artistPreviewUrl);
    }

    private void addVideoViewFragment(ArtistVideoViewFragment fragment, String artistPreviewUrl) {

        FragmentUtils.setArgsToFragment(fragment, URL,artistPreviewUrl);

        FragmentUtils.insertFragmentToActivity(R.id.fragment_container,getSupportFragmentManager(),fragment,ArtistVideoViewFragment.TAG);
    }

}

File: app/src/main/java/jp/co/geisha/itunestracker/service/view/activity/VideoViewActivity.kt
package jp.co.geisha.itunestracker.service.view.activity

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity

import jp.co.geisha.itunestracker.R
import jp.co.geisha.itunestracker.service.util.FragmentUtils
import jp.co.geisha.itunestracker.service.view.fragment.ArtistVideoViewFragment

import jp.co.geisha.itunestracker.service.ARTIST_VIDEO_VIEW_FRAGMENT_TAG
import jp.co.geisha.itunestracker.service.URL

class VideoViewActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_video_view)

        val artistPreviewUrl = intent.getStringExtra(URL)
        addVideoViewFragment(ArtistVideoViewFragment(), artistPreviewUrl)
    }

    private fun addVideoViewFragment(fragment: ArtistVideoViewFragment, artistPreviewUrl: String) {
        FragmentUtils.run {
            setArgsToFragment(fragment, URL, artistPreviewUrl)
            insertFragmentToActivity(R.id.fragment_container, supportFragmentManager, fragment, ARTIST_VIDEO_VIEW_FRAGMENT_TAG)
        }
    }
}
